### PR TITLE
Heat code perf tests

### PR DESCRIPTION
**Pull request by jeremiah-corrado**
Adds some tests to track the performance of a few versions of a 1D heat/diffusion finite difference solver.

These are all single locale 3-point stencil solvers that use a few different strategies for parallelism and synchronization.

- [x] test passing
- [x] tests passing w/ `-performance`

### CODE REVIEW

File: test/studies/lsuHeat/summary.chpl [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+const SOLUTION = if perfTest then 2.886724e+06 else 2.88666e+05;
+
+proc summarize(s: stopwatch, data: [] real) {
+  if perfTest then writeln("elapsed: ", s.elapsed());
+  const mean = + reduce data / (data.size: real);
+  const stdDev = sqrt(+ reduce ((data - mean) ** 2) / data.size: real);
+
+  // writef("stdDev: %.6dr, gt: %.6dr\n", stdDev, SOLUTION);
+  writeln((abs(stdDev - SOLUTION) / SOLUTION) < 1e-4);
+}
```
Comment (bmcdonald3) [inline test/studies/lsuHeat/summary.chpl:13]: How come we've no tested this one?

**Review by bmcdonald3 — APPROVED** [review]
Graph files look good to me, just one question abotu a notest

File: test/studies/lsuHeat/summary.chpl (hunk 1 above)
Comment (jeremiah-corrado) [reply test/studies/lsuHeat/summary.chpl:13]: The summarize proc used by the other tests to generate output, but this isn't testing anything directly

Comment (bmcdonald3) [reply test/studies/lsuHeat/summary.chpl:13]: Ah, OK, I see. Alright, well, this PR's got my seal of approval

**Pull request merged**